go/service/gregor.go

```diff
@@ -549,22 +549,29 @@ func (g *gregorHandler) pushStateThread(shutdownCh chan struct{}) {
 	}
 }
 
-func (g *gregorHandler) pushState(r keybase1.PushReason) {
+func (g *gregorHandler) pushStateOnce(r keybase1.PushReason) {
 	s, err := g.getState(context.Background())
 	if err != nil {
 		g.Warning(context.Background(), "Cannot push state in firehose handler: %s", err)
 		return
 	}
-	g.pushStateCh <- func() {
-		g.iterateOverFirehoseHandlers(func(h libkb.GregorFirehoseHandler) {
-			g.Debug(context.Background(), "pushState: pushing state with %d items", len(s.Items_))
-			h.PushState(s, r)
-		})
-		// Only send this state update on reception of new data, not a reconnect since we will
-		// be sending that on a different code path altogether (see OnConnect).
-		if g.badger != nil && r != keybase1.PushReason_RECONNECTED {
-			g.badger.PushState(context.Background(), s)
-		}
+	g.iterateOverFirehoseHandlers(func(h libkb.GregorFirehoseHandler) {
+		g.Debug(context.Background(), "pushState: pushing state with %d items", len(s.Items_))
+		h.PushState(s, r)
+	})
+	// Only send this state update on reception of new data, not a reconnect since we will
+	// be sending that on a different code path altogether (see OnConnect).
+	if g.badger != nil && r != keybase1.PushReason_RECONNECTED {
+		g.badger.PushState(context.Background(), s)
+	}
+}
+
+func (g *gregorHandler) pushState(r keybase1.PushReason) {
+	switch r {
+	case keybase1.PushReason_RECONNECTED, keybase1.PushReason_NONE:
```

Comment:
when does `PushReason_NONE` happen? does it ever? why not debounce?

Reply:
Unused, and I have `NEW_DATA` above as the reason, didn't want to muddle it.